core/server/web/shared/middlewares/trigger-webhooks.js

```diff
@@ -0,0 +1,13 @@
+const webhookService = require('../../../services/webhooks');
+const SITE_CHANGE_EVENT = 'site_changed';
+
+module.exports = function triggerWebhooks(req, res, next) {
+    res.on('finish', function triggerWebhookEvents() {
+        if (res.get('X-Cache-Invalidate') === '/*') {
+            webhookService.trigger(SITE_CHANGE_EVENT, {});
```

Comment:
@kirrg001 An alternative to directly triggering webhooks from here via service, is to do via event emitter pattern - `common.events.emit(SITE_CHANGE_EVENT)` - and webhooks `listen.js` service handles it and uses this triggers service to send requests. 

Pro is its more consistent with how currently webhooks are triggered and should be maybe in long term, con is this would lead to lot of changes in `listen.js` which is used in v0.1, like v0.1 tests will need update, as well as `trigger` method in v0.1 webhooks controller [here](https://github.com/TryGhost/Ghost/blob/master/core/server/api/v0.1/webhooks.js#L136) becomes redundant as this is only place its used. Does it make sense to do all these changes for longer consistency?

Reply:
The only place I can see `listen` used is here: https://github.com/TryGhost/Ghost/blob/master/core/server/index.js#L42 I would be surprised if you needed to change anything other than deleting the `trigger` method in v0.1 - but it would require some changes in listen, mainly around payload creation i guess?

Reply:
Also tests [here](https://github.com/rishabhgrg/Ghost/blob/master/core/test/unit/services/webhooks_spec.js#L29-L58) will need an update. But yeah overall agreed its doable and better approach, as long as we are ok v0.1 changes here 👻

Reply:
Yeah event sounds good 👍